api/src/services/utils/SessionUtility.ts

```diff
@@ -0,0 +1,67 @@
+import * as jsonwebtoken from "jsonwebtoken";
+import UserSession from "@common/models/resources/UserSession";
+import UserModel from "@common/models/resources/UserModel";
+import DateUtility from "@/services/utils/DateUtility";
+import RcdaAuthorizationPolicy from "@common/system/RcdaAuthorizationPolicy";
+import { RcdaRoles } from "@common/system/RcdaRoles";
+
+export default class SessionUtility {
+    constructor(private jwt: typeof jsonwebtoken, private dateUtility: DateUtility) {}
+
+    static getInstance() {
+        return new SessionUtility(jsonwebtoken, DateUtility.getInstance());
+    }
+
+    static jwtSignature = "54376454frwcbyx6c4wgurwj";
+
+    public getUserSession(user: UserModel): UserSession {
+        return {
+            userId: user.id,
+            roles: user.permissions.roles,
+            issued: this.dateUtility.currentDateString()
+        };
+    }
+
+    public getSessionToken(userSession: UserSession): string {
+        
+        return this.jwt.sign(userSession, SessionUtility.jwtSignature);
```

Comment:
Can definitley use SHA512 for the signature. We do have expiration login in place, but it is hardcoded to a specific date. Can move this into config. Can also look into using a refresh token to extend the session so that we can use shorter auth sessions.